commit 6b07342e25961dbd483671c130cc85603462806a
Author: Maxime Ripard <maxime.ripard@bootlin.com>
Date:   Thu May 17 15:37:59 2018 +0200

    drm/vc4: plane: Expand the lower bits by repeating the higher bits
    
    [ Upstream commit 3257ec797d3a8c5232389eb1952d4451e80f3931 ]
    
    The vc4 HVS uses an internal RGB888 representation of the frames, and will
    by default expand formats using a lower depth using zeros.
    
    This causes an issue when we try to use other compositing software such as
    pixman that fill the missing bits by repeating the higher significant bits.
    As such, we can't check the display output in a reliable way by doing a
    software composition and an hardware one and compare both.
    
    To prevent this, force the same behaviour so that we can do such things.
    
    Signed-off-by: Maxime Ripard <maxime.ripard@bootlin.com>
    Signed-off-by: Eric Anholt <eric@anholt.net>
    Link: https://patchwork.freedesktop.org/patch/msgid/20180517133759.25626-1-maxime.ripard@bootlin.com
    Signed-off-by: Sasha Levin <alexander.levin@microsoft.com>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>

diff --git a/drivers/gpu/drm/vc4/vc4_plane.c b/drivers/gpu/drm/vc4/vc4_plane.c
index 3363451b992b..203ddf5723e8 100644
--- a/drivers/gpu/drm/vc4/vc4_plane.c
+++ b/drivers/gpu/drm/vc4/vc4_plane.c
@@ -543,6 +543,7 @@ static int vc4_plane_mode_set(struct drm_plane *plane,
 	/* Control word */
 	vc4_dlist_write(vc4_state,
 			SCALER_CTL0_VALID |
+			VC4_SET_FIELD(SCALER_CTL0_RGBA_EXPAND_ROUND, SCALER_CTL0_RGBA_EXPAND) |
 			(format->pixel_order << SCALER_CTL0_ORDER_SHIFT) |
 			(format->hvs << SCALER_CTL0_PIXEL_FORMAT_SHIFT) |
 			VC4_SET_FIELD(tiling, SCALER_CTL0_TILING) |
diff -uN a/1.txt b/1.txt
--- a/dummy/rpi_1217_6b07342e25961dbd483671c130cc85603462806a.txt	1970-01-01 00:00:00.000000000 +0000
+++ b/dummy/rpi_1217_6b07342e25961dbd483671c130cc85603462806a.txt	2013-12-23 04:07:40.000000000 +0000
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
